src/featbin/paste-vectors.cc

```diff
@@ -0,0 +1,129 @@
+// featbin/paste-vectors.cc
+
+// Copyright 2020 Ivan Medennikov (STC-innovation Ltd)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "matrix/kaldi-matrix.h"
+#include "util/common-utils.h"
+
+namespace kaldi {
+
+void AppendVectors(const std::vector<Vector<BaseFloat> > &in,
+                   Vector<BaseFloat> *out) {
+  // Check the lengths
+  int32 tot_dim = in[0].Dim();
+  for (int32 i = 1; i < in.size(); ++i) {
+    int32 dim = in[i].Dim();
+    tot_dim += dim;
+  }
+  out->Resize(tot_dim);
+  int32 dim_offset = 0;
+  for (int32 i = 0; i < in.size(); ++i) {
+    int32 this_dim = in[i].Dim();
+    out->Range(dim_offset, this_dim).CopyFromVec(
+        in[i].Range(0, this_dim));
+    dim_offset += this_dim;
+  }
+}
+
+
+}  // namespace kaldi
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace std;
+
+    const char *usage =
+        "Concatenate vector files \n"
+        "Usage: paste-vectors <in-rspecifier1> <in-rspecifier2> [<in-rspecifier3> ...] <out-wspecifier>\n"
+        "See also: paste-feats, copy-feats, copy-matrix, append-vector-to-feats, concat-feats\n";
+
+    ParseOptions po(usage);
+
+    po.Read(argc, argv);
+
+    if (po.NumArgs() < 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    if (ClassifyRspecifier(po.GetArg(1), NULL, NULL)
+        != kNoRspecifier) {
+      // We're operating on tables, e.g. archives.
```

Comment:
This program can only work on archives. This `if` closes, and the program exists silently if this condition is false.

Reverse the condition and print an error, e.g. 
```suggestion
    if (ClassifyRspecifier(po.GetArg(1), NULL, NULL) != kNoRspecifier) {
      KALDI_ERR << "This program can operate only on tables, not on rxfiles";
    }
```
and then unindent the rest. `KALDI_ERR` will exits the program.